photon-ml/src/main/scala/com/linkedin/photon/ml/model/Coefficients.scala

```diff
@@ -51,12 +52,18 @@ protected[ml] case class Coefficients(means: Vector[Double], variancesOption: Op
   }
 
   override def equals(that: Any): Boolean = {
+
+    def cmp(a1: Vector[Double], a2: Vector[Double]): Boolean =
+      (a1.toArray zip a2.toArray).forall {
+        case ((m1, m2)) => math.abs(m2 - m1) < MathConst.HIGH_PRECISION_TOLERANCE_THRESHOLD
+      }
+
     that match {
       case other: Coefficients =>
-        val sameMeans = this.means.equals(other.means)
+        val sameMeans = cmp(this.means, other.means)
         val sameVariance =
           (this.variancesOption.isDefined && other.variancesOption.isDefined &&
-              this.variancesOption.get.equals(other.variancesOption.get)) ||
+              cmp(this.variancesOption.get, other.variancesOption.get)) ||
```

Comment:
I think we should also give this a more descriptive name and move it to VectorUtils.